AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/key-agreement-scheme.smithy

```diff
@@ -0,0 +1,72 @@
+namespace aws.cryptography.materialProviders
+
+use aws.polymorph#javadoc
+
+// Key Agreement Schemes
+@javadoc("Supported ECDH KeyAgreement Schemes")
+union KeyAgreementScheme {
+    StaticConfiguration: StaticConfigurations
+}
+
+@javadoc("Supported configurations for the StaticConfiguration Key Agreement Scheme")
+union StaticConfigurations {
+    AWS_KMS_ECDH: KmsEcdhStaticConfigurations,
+    RAW_ECDH: RawEcdhStaticConfigurations
+}
+
+@javadoc("Allowed configurations when using a KmsEcdhStaticConfigurations")
+union KmsEcdhStaticConfigurations {
+    KmsPublicKeyDiscovery: KmsPublicKeyDiscoveryInput,
+    KmsPrivateKeyToStaticPublicKey: KmsPrivateKeyToStaticPublicKeyInput,
+}
+
+@javadoc("List of configurations when using a RawEcdhStaticConfigurations")
+union RawEcdhStaticConfigurations {
+    PublicKeyDiscovery: PublicKeyDiscoveryInput,
+    RawPrivateKeyToStaticPublicKey: RawPrivateKeyToStaticPublicKeyInput,
+    EphemeralPrivateKeyToStaticPublicKey: EphemeralPrivateKeyToStaticPublicKeyInput 
+}
+
+
+@javadoc("Inputs for creating a KmsPublicKeyDiscovery Configuration. This is a DECRYPT ONLY configuration")
+structure KmsPublicKeyDiscoveryInput {
+    @required
+    @javadoc("AWS KMS key identifier belonging to the recipient")
+    recipientKmsIdentifier: KmsKeyId
+}
+
+@javadoc("Inputs for creating a KmsPrivateKeyToStaticPublicKey Configuration.")
+structure KmsPrivateKeyToStaticPublicKeyInput {
+    @required
+    @javadoc("AWS KMS Key Identifier belonging to the sender")
+    senderKmsIdentifier: KmsKeyId,
+    @javadoc("Sender Public Key. This is the raw public ECC key in DER format that belongs to the senderKmsIdentifier.")
+    senderPublicKey: Blob,
+    @required
+    @javadoc("Recipient Public Key. This is a raw public ECC key in DER format.")
```

Comment:
nit: I prefer "This MUST be a raw public ECC key in DER format" if that is a constraint we require here.